pkg/clusteragent/admission/mutate/auto_instrumentation_util.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+// Package mutate implements the mutations needed by the auto-instrumentation feature.
+package mutate
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// getOwnerNameAndKind returns the name and kind of the first owner of the pod if it exists
+// if the first owner is a replicaset, it returns the name
+func getOwnerNameAndKind(pod *corev1.Pod) (string, string, bool) {
+	owners := pod.GetOwnerReferences()
+
+	if len(owners) == 0 {
+		return "", "", false
+	}
+
+	owner := owners[0]
+	ownerName, ownerKind := owner.Name, owner.Kind
+
+	if ownerKind == "ReplicaSet" {
+		deploymentName := kubernetes.ParseDeploymentForReplicaSet(ownerName)
+		if deploymentName != "" {
+			ownerKind = "Deployment"
+			ownerName = deploymentName
+		}
+	}
+
+	return ownerName, ownerKind, true
+}
+
+func getLibListFromDeploymentAnnotations(store workloadmeta.Store, deploymentName, ns, registry string) []libInfo {
+	libList := []libInfo{}
+
+	// populate libInfoList using the languages found in workloadmeta
+	id := fmt.Sprintf("%s/%s", ns, deploymentName)
+	deployment, err := store.GetKubernetesDeployment(id)
+	if err != nil {
+		return libList
+	}
+
+	for containerName, languages := range deployment.ContainerLanguages {
+		for _, lang := range languages {
+			imageToInject := fmt.Sprintf(imageFormat, registry, lang.Name, "latest")
```

Comment:
I think we might prefer a function. What do you think ?

Reply:
You mean a function to calculate the image to be be injected?

Reply:
To determine the image name and tag yes

Reply:
I agree.

I just followed the same pattern used in the existing implementation of the admission controller.

I will modify it by having a separate function for this to improve code structure. 

Thanks